starknetdata/feeder/feeder.go

```diff
@@ -113,3 +113,76 @@ func (f *Feeder) StateUpdate(ctx context.Context, blockNumber uint64) (*core.Sta
 func (f *Feeder) StateUpdatePending(ctx context.Context) (*core.StateUpdate, error) {
 	return f.stateUpdate(ctx, "pending")
 }
+
+func (f *Feeder) stateUpdateWithBlock(ctx context.Context, blockID string) (*core.StateUpdate, *core.Block, error) {
+	response, err := f.client.StateUpdateWithBlock(ctx, blockID)
+	if err != nil {
+		// TODO: remove once the new feeder is available
+		if err.Error() == "400 Bad Request" {
```

Comment:
I feel like we should be to use `errors.Is` here, but maybe string matching is necessary.

Reply:
I guess string matching is necessary here. But like the first review, this check will be removed in the next commit.